pilot/pkg/networking/core/configgen.go

```diff
@@ -35,7 +35,7 @@ type ConfigGenerator interface {
 	BuildListeners(node *model.Proxy, push *model.PushContext) []*listener.Listener
 
 	// BuildClusters returns the list of clusters for the given proxy. This is the CDS output
-	BuildClusters(node *model.Proxy, push *model.PushContext) []*cluster.Cluster
+	BuildClusters(node *model.Proxy, push *model.PushContext, w *model.WatchedResource) []*cluster.Cluster
```

Comment:
Passing some set of configs to generate makes sense, but `model.WatchedResource` may not be the best thing for delta XDS in the future. Maybe just a set of resources?

We can always change it for delta when needed though

Reply:
agree -- `model.WatchedResource` is exclusively for `DiscoveryRequest` and `DiscoveryResponse`

Reply:
agree -- model.WatchedResource is exclusively for DiscoveryRequest and DiscoveryResponse